lib/datasource/packagist/index.ts

```diff
@@ -38,6 +39,7 @@ interface RegistryFile {
 }
 interface RegistryMeta {
   files?: RegistryFile[];
+  providerPackages: Record<string, string>;
```

Comment:
This is now a required property?

Reply:
The variable would be created anyways in here: https://github.com/renovatebot/renovate/pull/6028/files#diff-9d4e78c90551425d7672fa30aec4d667L180

I moved it to the registry meta so we can extract providers at root level here: https://github.com/renovatebot/renovate/pull/6028/files#diff-9d4e78c90551425d7672fa30aec4d667R53-R55

Previously, providers would only be supported from the `provider-includes` files (see https://repo.packagist.org/packages.json) but they can exist in packages.json themselves (see https://composer.typo3.org/packages.json for an example) which would previously go unnoticed

Reply:
I was more referring to the fact that to create an empty `providerPackages` for any `RegistryMeta` object. Is it necessary? Like here:

https://github.com/renovatebot/renovate/blob/65a8f33ad9ba797e5cdeaeac9259f6ae90fb923d/lib/datasource/packagist/index.ts#L53-L55

Reply:
Alternative would be to create it only if necessary and later have a check and create it anyways. I just moved to creation to an earlier moment